app/2.0/docs/tools/build-for-production.md

```diff
@@ -0,0 +1,222 @@
+
+The Polymer CLI and `polymer-build` library support the following transforms:
+
+* [Splitting inline JavaScript and CSS from HTML files](#splitting)
+* [Minifying HTML, JavaScript and CSS](#minifying)
+* [Compiling ES6 to ES5](#compiling)
+* [Bundling resources to reduce the total number of HTTP requests made by the user's browser](#bundling)
+
+### Splitting inline JavaScript and CSS from HTML files {#splitting}
+  
+By default, the Polymer CLI build process uses a built in feature of `polymer-build` to split inline JavaScript and CSS from HTML files. The build process automatically splits these streams in order to process them appropriately - for example, for minification, compilation or both.
+
+By default, the Polymer build tools rejoin JavaScript and HTML after processing.
+
+To split your code from HTML in order to deploy into an environment that requires this, you can use [Crisper](https://github.com/PolymerLabs/crisper).
+  
+### Minifying HTML, JavaScript and CSS {#minifying}
+
+The Polymer build tools provide options to minify HTML, JavaScript and CSS by stripping whitespace and comments. You can use these on the command line, or configure them in `polymer.json`. See the [polymer.json specification](polymer-json) for more details.
+
+**Function**|**CLI flag**|**Entry in polymer.json**
+--- | --- | ---
+Minify inlined and external JavaScript | `--js-minify` | `"js": {"minify": true}` 
+Minify inlined and external CSS | `--css-minify` | `"css": {"minify": true}` 
+Minify HTML | `--html-minify` | `"html": {"minify": true}` 
+
+Example: Build a project. Use CLI options to minify JavaScript, CSS and HTML {.caption}
+
+```bash
+polymer build --js-minify --css-minify --html-minify 
+```
+
+Example: A build object in `polymer.json` that minifies JavaScript, CSS and HTML {.caption}
+
+```
+"builds": [{
+  "js": {"minify": true},
+  "css": {"minify": true},
+  "html": {"minify": true}
+}]
+```
+
+### Compiling ES6 to ES5 {#compiling}
+
+Polymer 2.x and its native elements are written using ES6, allowing class definitions, inheritance and modular code. Support for ES6 is required in order for a browser to implement the [custom elements](https://developers.google.com/web/fundamentals/getting-started/primers/customelements) specification.
+
+Because Polymer 2.0 uses ES6 and HTML Custom Elements, it is always best to serve ES6 to browsers with full ES6 support.
+
+These browsers fully support ES6:
+
+* Chrome or Chromium version 49 or later.
+* Opera 36 or later.
+* Safari or Mobile Safari 10 or later.
+* Edge 15.15063 or later.
+* Firefox 51 or later.
+
+You should serve compiled ES5 only to older browsers that don't support ES6.
+
+If you need to statically host your code and serve a single version to all browsers, however, you should compile all code to ES5. In this case, you can include a shim - a lightweight polyfill that lets compiled ES5 work on browsers that support native custom elements.
+  
+```bash
+polymer build --js-compile 
+```
```

Comment:
Show just flag and equivalent polymer.json snippet here, like with minification?